### PR TITLE
Add `/state/:projectId` endpoint

### PR DESCRIPTION
**Pull request by trizin**
Response example:

```json
{
  "level": "New Project",
  "ceiling": 3000,
  "projectCategory": "build",
  "grantsProposed": 1,
  "grantsReceived": 0,
  "grantsCompleted": 0,
  "availableEarmaks": [
    "newproject",
    "coretech",
    "general",
    "outreach"
  ]
}
```

### CODE REVIEW

File: backend/routes/project.js [hunk 1]
```diff
@@ -297,6 +297,71 @@ router.post("/proposal/list", function (req, res) {
...
+
+        if (grantsCompleted == 0) {
+          if (projectCategory == "outreach")
+            availableEarmaks.push("newprojectoutreach");
+          else availableEarmaks.push("newproject");
+        }
+
+        availableEarmaks.push("coretech");
+        availableEarmaks.push("general");
+        availableEarmaks.push("outreach");
```
Comment (KatunaNorbert) [inline backend/routes/project.js:350]: shouldn't this earmarks from above be available only if project has at least 1 proposal funded?

File: backend/routes/project.js [hunk 2]
```diff
@@ -297,6 +297,71 @@ router.post("/proposal/list", function (req, res) {
...
+            x["Proposal State"] === "Granted" ||
+            x["Proposal State"] === "Funded"
+        ).length;
+        const grantsCompleted = proposals.filter(
+          (x) => x.delivered.status == 2
+        ).length;
+
+        const projectCategory = project.projectCategory;
+
+        const level = levels(grantsCompleted);
```
Comment (KatunaNorbert) [inline backend/routes/project.js:338]: wouldn't it be more feasible to just get all the calculated values from above from the project summary table? I guess those are not cached yet but if we cache them we could reduce a lot of code.

File: backend/routes/project.js (hunk 1 above)
Comment (trizin) [reply backend/routes/project.js:350]: Right, thanks for pointing this out

**Comment by KatunaNorbert** [comment]
I think we could squeeze the changes for the new earmark in here, looks like this endpoint will dictate the earmark logic

**Comment by KatunaNorbert** [comment]
And we could also remove the earmark logic from create proposal:
https://github.com/oceanprotocol/oceandao-proposal-portal/blob/main/backend/routes/project.js#L80

**Review by KatunaNorbert — APPROVED** [review]
Looks good!

**Pull request merged**